pennylane/devices/default_qubit.py

```diff
@@ -216,6 +214,11 @@ def _apply_operation(self, state, operation):
         Returns:
             array[complex]: output state
         """
+
+        if hasattr(self, "_torch_device"):
```

Comment:
Instead of changing this here, we could extend the operation in the torch device:

```python
def _apply_operation(self, state, operation):
    if state.device != self._torch_device:
        state = state.to(self._torch_device
    return super()._apply_operation(self, state, operation)
```

Or something like that.